browser.js

```diff
@@ -203,18 +203,18 @@ function wrap (opts, logger, level) {
 
 function asObject (logger, level, args, ts) {
   if (logger._serialize) applySerializers(args, logger._serialize, logger.serializers, logger._stdErrSerialize)
-  var msg = args[0]
+  var argsCloned = args.slice()
```

Comment:
we need to check the `transmit` functionality alongside `asObject` - because it takes in the `args` array (after modification) - that might be a bug which you've fixed or you might have introduced a bug by cloning.. I'm not sure without checking. Would you mind adding a test for transmit + asObject true with these changes?

Reply:
Sorry im not sure what you mean here. 

`transmit` is an object key, but `send` is the function. The three test cases I added cover all the permutations of `send` being called with the correct messages. Are you thinking we need to expand the test cases for `write`? Because we have cases covering that `send` actually gets called with the correct info under the different conditions.

Reply:
Perhaps you mean you want to make sure that `write` is being called correctly in the `asObject` scenario?

```js
if (opts.asObject) write.call(proto, asObject(this, level, args, ts))
else write.apply(proto, args)
```

.i.e. `asObject` is returning the correct formatted data?

Reply:
let me know if this is what you are looking for, or something different: https://github.com/pinojs/pino/pull/590/commits/139266aebfdc95b73ecca4755bbdc752a08c72f8

Reply:
based on your description, that's precisely what I covered in the original set of tests I added, because thats what was **broken** prior to this change:

```js
test('passes send function message strings in logEvent object when asObject is set', ({ end, same, is }) => {
  const logger = pino({
    browser: {
      asObject: true,
      write: noop,
      transmit: {
        send (level, { messages }) {
          is(messages[0], 'test')
          is(messages[1], 'another test')
        }
      }
    }
  })

  logger.fatal('test', 'another test')

  end()
})

test('passes send function message objects in logEvent object when asObject is set', ({ end, same, is }) => {
  const logger = pino({
    browser: {
      asObject: true,
      write: noop,
      transmit: {
        send (level, { messages }) {
          same(messages[0], { test: 'test' })
          is(messages[1], 'another test')
        }
      }
    }
  })

  logger.fatal({ test: 'test' }, 'another test')

  end()
})
```
In these scenarios, both `transmit` and `asObject` are set, and that is what was breaking before this PR (in the case of just strings).

What am I missing here?

Reply:
you're not missing anything, it's me thanks @justinhelmer

Reply:
👍